test/metabase/lib/expression_test.cljc

```diff
@@ -549,15 +549,36 @@
                                                          100)
                                                   nil))))))
 
-(deftest ^:parallel diagnose-expression-literals-test
-  (testing "top-level literals are not allowed"
-    (let [query (lib/query meta/metadata-provider (meta/table-metadata :orders))
+(deftest ^:parallel diagnose-expression-literals-without-feature-support-test
+  (testing "top-level literals are not allowed if the :expression-literals feature is not supported"
+    (let [metadata-graph (.-metadata-graph meta/metadata-provider)
+          restricted-metadata-graph (update metadata-graph :features disj :expression-literals)
+          restricted-provider (meta.graph-provider/->SimpleGraphMetadataProvider restricted-metadata-graph)
```

Comment:
Not necessarily for this PR, but I've seen similar things elsewhere. It feels like a helper to update the `:features` on a fake `MetadataProvider`'s database would be handy. Probably write it in `update` fashion, since sometimes we're adding things and sometimes removing them.